### PR TITLE
CYS - set flowType when the state machine is instantiated

### PR DESCRIPTION
**Pull request by gigitux**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that you are on a WooCommerce site where the CYS AI flow works.
2. Open the network tools.
3. Head over to WooCommerce > Home > Customize your store.
4. Ensure that the `status.json` is fetched.
5. Click on "Design With AI" and proceed to the Pattern Assembler.
6. Ensure that the `status.json` isn't fetched again. You should see just one entry:

image

7. Smoke test the CYS flow.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

CYS - set flowType when the state machine is instantiated.

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/customize-store/intro/services.ts [hunk 1]
```diff
@@ -164,4 +166,31 @@ export const setFlags = async () => {
 		// all of them to resolve before returning.
 		await Promise.all( Object.values( _featureFlags ) );
 	}
+
+	// Set FlowType flag
+	if ( isWooExpress() ) {
+		try {
+			const { status } = await fetchAiStatus();
```
Comment (nefeline) [inline plugins/woocommerce-admin/client/customize-store/intro/services.ts:173]: At what point is the state machine instantiated? Is it instantiated just once? Have you considered any performance impacts when making the GET request to openAI on instantiation?

Comment (gigitux) [reply plugins/woocommerce-admin/client/customize-store/intro/services.ts:173]: > At what point is the state machine instantiated?

The state machine is only instantiated when the intro page is loaded. Technically, in our case, the state machine is instantiated twice for the iframe issue, but we avoid the prefetch of the `status.json` via https://github.com/woocommerce/woocommerce/pull/46528/files#diff-3c46256a36068561a283581049ff32247c4509775d558bf5c888f5b840ad8e33R171

**Comment by yansern** [comment]
I can see `status.json` is only called once, so this is good now.
image

And also I am able to land directly in `/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fdesign-with-ai` now.

Thank you so much for the hard work. Didn't think it would involve quite a bit of changes. :)

File: plugins/woocommerce-admin/client/customize-store/intro/services.ts (hunk 1 above)
Comment (nefeline) [reply plugins/woocommerce-admin/client/customize-store/intro/services.ts:173]: Alright, thanks! Follow-up questions: did you test this change with Jetpack disabled? Will users be able to proceed to the CYS AI offline flow without any problems with this change in place? Are we accounting for user consent? Ideally, we should not be making requests to AI if the user didn't grant us consent during initial plugin setup.

**Review by nefeline — COMMENTED** [review]
@gigitux great job here! I've left a couple of questions as I want to make sure we are not bypassing user consent and/or introducing regressions whenever Jetpack connection is disabled: after clearing those out, we should be all set 🙌

File: plugins/woocommerce-admin/client/customize-store/intro/services.ts (hunk 1 above)
Comment (gigitux) [reply plugins/woocommerce-admin/client/customize-store/intro/services.ts:173]: > Will users be able to proceed to the CYS AI offline flow without any problems with this change in place?

If you mean the No AI flow, this code runs only on Woo Express.

Also, I tested this change simulating the AI offline and everything works as expected. 

> Are we accounting for user consent? Ideally, we should not be making requests to AI if the user didn't grant us consent during initial plugin setup.

Given that this code runs only on Woo Express, I guess that we're fine on this point. Futhermore, even before this refactor, we didn't do this kind of check.

Comment (nefeline) [reply plugins/woocommerce-admin/client/customize-store/intro/services.ts:173]: > If you mean the No AI flow, this code runs only on Woo Express.

I mean the AI offline flow, which runs on Woo Express: https://github.com/woocommerce/woocommerce/pull/41656

> I tested this change simulating the AI offline and everything works as expected.

Thanks for confirming.

**Review by nefeline — APPROVED** [review]

**Comment by albarin** [comment]
Hey @gigitux, good job! Works as expected.
I think some of the guards are not used anymore, like `isWooExpress` or `isNotWooExpress`, should we remove them?

**Comment by gigitux** [comment]
> I think some of the guards are not used anymore, like isWooExpress or isNotWooExpress, should we remove them?

Good point! I removed them with f2d1059b630424438f4b8a105257177a7b2b68ae.

**Review by albarin — APPROVED** [review]

**Pull request merged**